### PR TITLE
[proc] made endpoints and api keys match length instead of returning an error

### PR DESCRIPTION
**Pull request by shang-wang**
Added logics to match the length of endpoints and api keys instead of returning an error. @DataDog/burrito

### CODE REVIEW

File: config/config_test.go [hunk 1]
```diff
@@ -163,24 +163,45 @@ func TestGetHostname(t *testing.T) {
...
 	agentConfig, err = NewAgentConfig(configFile, nil)
+	assert.NoError(err)
+	assert.Equal(2, len(agentConfig.APIEndpoints))
 	assert.Equal("foo", agentConfig.APIEndpoints[0].APIKey)
 	assert.Equal("process.datadoghq.com", agentConfig.APIEndpoints[0].Endpoint.Hostname())
 	assert.Equal("bar", agentConfig.APIEndpoints[1].APIKey)
 	assert.Equal("process.datadoghq.eu", agentConfig.APIEndpoints[1].Endpoint.Hostname())
+
+	// if endpoint count is greater than api_key count, drop additional endpoints
+	ddAgentConf, _ = ini.Load([]byte(strings.Join([]string{
```
Comment (conorbranagan) [inline config/config_test.go]: Can we add a test for the reverse case? (More api keys than endpoints)

Comment (shang-wang) [reply config/config_test.go]: Well, the first test case in this test function is already doing that. If you don't provide any endpoints there's actually a default endpoint, so it's 2 api_keys and 1 endpoint.

Comment (shang-wang) [reply config/config_test.go]: But I should add some comment.

Comment (shang-wang) [reply config/config_test.go]: Oh wait, there's already comment :)

**Pull request merged**